### DISCUSSION TITLE
onBlur onFocus events

### DISCUSSION THREAD

**Discussion by mwx23** [lona-web-org/lona#169] [Q&A]
I've got a Widget that when the user clicks it, it becomes an input, and when the focus has left (i.e. onBlur) I'd like to revert it.

Or is this something I would do in a custom frontend widget?

Are there any plans to include more events like onBlur and onFocus?

**Comment by fscherf** [comment]
Hi @mwx23, 

What does "when the user clicks, it becomes an input" mean? What is the widget before and after?

Blur and focus events are actually a pretty good idea. Are there more events that would make sense to implement?

**Comment by mwx23** [comment]
I've now just realised how to get around it without a onblur event. I'll have a go and report back tomorrow.

I just had a look at https://www.w3schools.com/tags/ref_eventattributes.asp 

ondrag* + ondrop - this is my personal highest priority
onkey* - keyboard shortcuts :)
ondblclick - not sure about this, trying to be touch friendly
oncontextmenu - not good for touch devices but would love this on desktop

**Reply by fscherf** [reply]
Ok. ``blur``, ``focus`` and ``hover`` are pretty simple to implement. I will start working on them after #170 is finished. Do you want to send a PR for drag and drop?

**Reply by mwx23** [reply]
I'm happy to have a go at drag and drop. It will force me to learn how `lona` works!
